samza-core/src/main/java/org/apache/samza/clustermanager/ClusterBasedJobCoordinator.java

```diff
@@ -264,9 +279,15 @@ public void run() {
       partitionMonitor.ifPresent(StreamPartitionCountMonitor::start);
       inputStreamRegexMonitor.ifPresent(StreamRegexMonitor::start);
 
+      // ContainerPlacementHandler thread has to start after the cpm is started
+      LOG.info("Starting the container placement handler thread");
+      containerPlacementUtil.start();
+      containerPlacementHandlerThread.start();
+
       boolean isInterrupted = false;
 
-      while (!containerProcessManager.shouldShutdown() && !checkAndThrowException() && !isInterrupted) {
+      while (!containerProcessManager.shouldShutdown() && !checkAndThrowException() && !isInterrupted
+          && checkContainerPlacementHandlerThreadIsAlive()) {
```

Comment:
nit: I personally prefer to have a new line for each condition if they can not fit in one line:

!containerProcessManager.shouldShutdown()
&& !checkAndThrowException()
&& !isInterrupted
&& checkContainerPlacementHandlerThreadIsAlive()